org.idempiere.test/src/org/idempiere/test/base/MatchInvTest.java

```diff
@@ -0,0 +1,204 @@
+package org.idempiere.test.base;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+
+import org.compiere.acct.Doc;
+import org.compiere.acct.DocManager;
+import org.compiere.model.MAccount;
+import org.compiere.model.MAcctSchema;
+import org.compiere.model.MBPartner;
+import org.compiere.model.MClientInfo;
+import org.compiere.model.MDocType;
+import org.compiere.model.MFactAcct;
+import org.compiere.model.MInOut;
+import org.compiere.model.MInOutLine;
+import org.compiere.model.MInvoice;
+import org.compiere.model.MInvoiceLine;
+import org.compiere.model.MMatchInv;
+import org.compiere.model.MOrder;
+import org.compiere.model.MOrderLine;
+import org.compiere.model.MProduct;
+import org.compiere.model.MRMA;
+import org.compiere.model.MRMALine;
+import org.compiere.model.MWarehouse;
+import org.compiere.model.ProductCost;
+import org.compiere.process.DocAction;
+import org.compiere.process.DocumentEngine;
+import org.compiere.process.ProcessInfo;
+import org.compiere.util.Env;
+import org.compiere.wf.MWorkflow;
+import org.idempiere.test.AbstractTestCase;
+import org.junit.jupiter.api.Test;
+
+public class MatchInvTest extends AbstractTestCase {
+
+	public MatchInvTest() {
+	}
+	
+	@Test
+	/**
+	 * https://idempiere.atlassian.net/browse/IDEMPIERE-4173
+	 */
+	public void testMatShipmentPosting() {
```

Comment:
if possible, please add test case for normal material receipt posting to prove that the changes doesn't break normal posting for po - mr -invoice matching.